bug: canonical issue-sync caller template uses `secrets: inherit` — fails to pass SYNC_PAT to reusable workflow on cross-account consumers

## Description

The canonical issue-sync caller template at `.agents/templates/workflows/issue-sync-caller.yml` uses `secrets: inherit` to pass `SYNC_PAT` to the reusable workflow. For consumers in a **different account/org from `marcusquinn`** (which is every downstream user — this is a public framework), `secrets.SYNC_PAT` evaluates to an empty string inside the reusable workflow at runtime, even when the secret is correctly set on the caller repo and the reusable workflow declares it as an optional secret input.

Result: every push/issue/PR-close event triggers an issue-sync run that warns "SYNC_PAT not set", falls back to `GITHUB_TOKEN`, and is rejected by branch protection with `GH006: Protected branch update failed`. Operators receive a cascade of run-failure emails and assume their secret is misconfigured — re-set it, see the same failure, file a ticket. The framework's own `marcusquinn/aidevops` repo is the only one where this works, because there it's same-repo `secrets: inherit`.

## Sister-bug context

This surfaced immediately after #20967 (caller template missing `permissions:` block) was fixed via PR #2087 in `robstiles/qs-agency`. The `permissions:` fix made the workflow actually start; this bug is what it now fails on. Both bugs share the same root cause class: the canonical caller template was authored against the framework's own self-test scenario (`marcusquinn/aidevops` calling itself) and assumes co-located, same-account semantics that don't hold for downstream consumers.

## Reproducer

**Symptom command** (any push to `TODO.md` on a downstream consumer repo, or manual dispatch):

```
gh workflow run "Issue Sync - Bi-directional TODO.md ↔ GitHub Issues" --repo robstiles/qs-agency -f command=push
```

**Actual output** (from `gh run view <id> --log-failed`):

```
##[warning]SYNC_PAT secret is not set — TODO.md auto-sync push will be rejected by branch protection (GH006).
##[warning]Fix: gh secret set SYNC_PAT --repo robstiles/qs-agency --body "<FINE_GRAINED_PAT>" (Contents: Read and write)
...
remote: error: GH006: Protected branch update failed for refs/heads/main.
remote: 
remote: - Required status check "review-bot-gate" is expected.
To https://github.com/robstiles/qs-agency
 ! [remote rejected] main -> main (protected branch hook declined)
error: failed to push some refs to 'https://github.com/robstiles/qs-agency'
All push attempts failed
##[error]Process completed with exit code 1.
```

**Expected output**: the push step uses `SYNC_PAT` (which IS set on the repo), bypasses branch protection via `enforce_admins: false`, and the TODO.md update lands on `main`.

**Causal evidence** that the secret IS set but unreachable:

- `gh secret list --repo robstiles/qs-agency` returns `SYNC_PAT  2026-04-19T22:06:10Z` (set, value present, not expired in storage)
- The runtime `Check SYNC_PAT visibility` step computes `SYNC_PAT_PRESENT: ${{ secrets.SYNC_PAT != '' && 'true' || '' }}` and consistently logs `SYNC_PAT_PRESENT: ` (empty) across every job (push-triggered, issue-triggered, manual `workflow_dispatch`)
- All 15+ touchpoints to `secrets.SYNC_PAT` inside `issue-sync-reusable.yml` (lines 71, 91, 117, 125, 140, 148, 178, 217, 233, 252, 300, 315, 332, 396, 789, 827) receive empty
- The user's prior local `issue-sync.yml` (pre-migration, t237 / commit `4b54bcb` in `robstiles/qs-agency`) had the **same** `${{ secrets.SYNC_PAT || secrets.GITHUB_TOKEN }}` expression INLINE in the same repo and worked correctly. The migration to the reusable-workflow pattern (PR #2083, generated by `aidevops sync-workflows --apply`) is when the secret resolution broke.

## Steps to Reproduce

1. In a repo NOT in the `marcusquinn` org/account, set `SYNC_PAT` as a repo secret (`gh secret set SYNC_PAT --repo <owner>/<repo>`)
2. Configure classic branch protection on `main` with at least one required status check, `enforce_admins: false`
3. Run `aidevops sync-workflows --apply` to install the canonical caller template (or use the equivalent `.github/workflows/issue-sync.yml` from `.agents/templates/workflows/issue-sync-caller.yml`)
4. Push a change to `TODO.md`
5. Observe: workflow runs, `SYNC_PAT_PRESENT` logs as empty, push step fails with GH006

## Why this matters

1. **Affects every downstream consumer.** The framework's own `marcusquinn/aidevops` repo is the only one where `secrets: inherit` works for this reusable workflow — because there it's same-repo. Every other consumer using the canonical template (per `aidevops check-workflows`) is silently broken.

2. **Misleading error message.** The reusable workflow's existing `Check SYNC_PAT visibility` step warns "SYNC_PAT secret is not set" — but in this scenario the secret IS set on the caller. Operators chase a phantom misconfiguration instead of the actual cross-account inherit limitation.

3. **`aidevops sync-workflows` will overwrite local fixes.** A consumer who manually patches their caller to use explicit `secrets: SYNC_PAT: ${{ secrets.SYNC_PAT }}` will have it stomped on the next `sync-workflows --apply` (which is also auto-suggested by `aidevops update`).

4. **Sister-bug pattern continues.** This is the second instance in 24 hours where the canonical caller template's framework-self-test assumption breaks for downstream consumers (see #20967). The class of bug deserves a structural mitigation, not just a per-bug patch.

## Considerations for the fix

The framework devs to choose the path. Some options to consider — these are not exhaustive:

**Option A: switch the canonical template to explicit secret pass-through**

```yaml
jobs:
  sync:
    uses: marcusquinn/aidevops/.github/workflows/issue-sync-reusable.yml@main
    secrets:
      SYNC_PAT: ${{ secrets.SYNC_PAT }}
    with:
      command: ${{ inputs.command || '' }}
```

- Works for cross-account/cross-org consumers
- Requires the template to enumerate every secret the reusable workflow declares; adding a new optional secret upstream becomes a breaking change for callers (they need to resync)
- `secrets:` and `secrets: inherit` cannot be mixed in the same job — choosing this means committing to explicit enumeration forever

**Option B: keep `secrets: inherit` but document the cross-account limitation**

- Document in `reference/reusable-workflows.md` that `secrets: inherit` requires same-org/same-enterprise membership
- Suggest consumers in different orgs use a forked / vendored copy of the reusable workflow
- Doesn't fix the bug; just shifts the burden

**Option C: better pre-flight diagnostic**

- The reusable workflow's existing "SYNC_PAT not set" warning is wrong in this scenario (the secret IS set on the caller, just not reaching the called workflow)
- Detect the cross-account case (caller-side secret present indicator vs called-side empty) and emit a more actionable warning naming the inherit limitation
- Doesn't fix the bug, but reduces wasted-triage cost
- Would require the caller template to do a pre-flight visibility check before calling the reusable workflow — adds template surface area

**Option D: investigate whether GitHub Actions has changed `secrets: inherit` semantics**

- Worth confirming the actual current GitHub Actions behavior by reading current docs and running a controlled cross-org test before assuming the limitation is the cause
- Possibility: the limitation is real per docs, OR there's something specific to how the framework's reusable workflow declares the secret that's interacting with inherit (e.g., a quirk in how `required: false` is handled)
- A 30-min controlled test would confirm or rule this out before committing to Option A's enumeration tax

## Workarounds Applied

In the observing session (`robstiles/qs-agency`), no workaround has been applied yet — the user wanted the upstream issue filed first so the canonical fix lands in the framework rather than drifting locally. Candidate consumer-side workaround (subject to being stomped by next `sync-workflows --apply`):

```yaml
jobs:
  sync:
    uses: marcusquinn/aidevops/.github/workflows/issue-sync-reusable.yml@main
    secrets:
      SYNC_PAT: ${{ secrets.SYNC_PAT }}
    with:
      command: ${{ inputs.command || '' }}
```

## Cross-references

- Sister bug **#20967** (caller template missing `permissions:` block) — same template, same root cause class (framework self-test passes, downstream broken)
- Prior framework SYNC_PAT work: t2048 (#18643, fine-grained PAT path), t2166 (#19488, extend SYNC_PAT to all four jobs), t2696 (#20324, extract visibility step), t2806 (#20745, rulesets-based protection detection)
- Workflow drift system: t2778 (`aidevops check-workflows`), t2779 (`aidevops sync-workflows`) — the canonical template is what consumers will keep getting reset to
- Live reproduction case: `robstiles/qs-agency` (secret set, workflow runs, push fails — available for verification of any candidate fix)

## Files to inspect

- `.agents/templates/workflows/issue-sync-caller.yml` — canonical caller template (the file to fix)
- `.github/workflows/issue-sync-reusable.yml` — reusable workflow declaring the secret input (15+ touchpoints to `secrets.SYNC_PAT`)
- `.agents/scripts/sync-workflows-helper.sh` (or wherever `aidevops sync-workflows --apply` lives) — needs to ship the corrected template

## Acceptance criteria

1. Canonical caller template passes `SYNC_PAT` in a way that works for cross-account/cross-org consumers
2. `aidevops sync-workflows --apply` on a downstream consumer (different account from `marcusquinn`) produces a caller that successfully resolves `secrets.SYNC_PAT` inside the reusable workflow at runtime
3. Decision documented (in `reference/reusable-workflows.md` or equivalent) on `secrets: inherit` vs explicit pass-through and why
4. Diagnostic message in the reusable workflow updated so it doesn't claim "SYNC_PAT not set" when the actual issue is cross-account inherit failure
5. Regression test or guidance that catches the next "framework self-test passes, downstream broken" instance of this class

## Environment

- **aidevops version**: 3.11.5
- **Latest version**: 3.11.5
- **AI Assistant**: Claude Code
- **OS**: Ubuntu 24.04.4 LTS
- **Shell**: bash 5.2.21(1)-release
- **Observing repo**: robstiles/qs-agency (main)
- **gh CLI**: gh version 2.91.0

<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.11.5 plugin for [OpenCode](https://opencode.ai) v1.14.25 with claude-opus-4-6 spent 23m and 24,942 tokens on this with the user in an interactive session.

--- a/.agents/reference/reusable-workflows.md
+++ b/.agents/reference/reusable-workflows.md
@@ -95,7 +95,8 @@ jobs:
     uses: marcusquinn/aidevops/.github/workflows/issue-sync-reusable.yml@v3.9.0
     #                                                                   ^^^^^^^
     #                                                                   change this
-    secrets: inherit
+    secrets:
+      SYNC_PAT: ${{ secrets.SYNC_PAT }}
 ```
 
 Keep pinned callers in sync with aidevops releases via:
@@ -109,7 +110,29 @@ See also [`auto-dispatch.md`](auto-dispatch.md) for the `SYNC_PAT` requirement (
 
 ## Security model
 
-- **`secrets: inherit`** in the caller grants the reusable workflow access to all the caller's secrets. This is intentional — `SYNC_PAT` is the only secret the reusable workflow needs, and it varies per-repo. `inherit` is the simplest way to expose it without listing every secret by name.
+- **`secrets: inherit` only works within the same GitHub account/org (GH#20976).** Cross-account callers (every downstream user — `marcusquinn/aidevops` is the only same-account consumer) receive empty values for caller-repo secrets when using `secrets: inherit` against a reusable workflow in a different account. Explicit secret pass-through is required:
+
+  ```yaml
+  # CORRECT for cross-account consumers (canonical template uses this):
+  jobs:
+    sync:
+      uses: marcusquinn/aidevops/.github/workflows/issue-sync-reusable.yml@main
+      secrets:
+        SYNC_PAT: ${{ secrets.SYNC_PAT }}
+
+  # BROKEN for cross-account consumers (SYNC_PAT resolves to empty):
+  jobs:
+    sync:
+      uses: marcusquinn/aidevops/.github/workflows/issue-sync-reusable.yml@main
+      secrets: inherit   # ← only works within the same account/org
+  ```
+
+  The canonical caller templates (`issue-sync-caller.yml`, `loc-badge-caller.yml`) use explicit pass-through. `review-bot-gate-caller.yml` and `maintainer-gate-caller.yml` keep `secrets: inherit` because they only reference `secrets.GITHUB_TOKEN` internally — `GITHUB_TOKEN` is always provided by the runner and is unaffected by the cross-account limitation.
+
+  If a new user-defined secret is added to a reusable workflow's `secrets:` input block, the matching caller template MUST also add it as an explicit `secrets: MySecret: ${{ secrets.MySecret }}` entry. This is the enumeration trade-off of explicit pass-through — adding a new optional secret upstream requires a template bump + `aidevops sync-workflows --apply` on consumers. It is preferable to silently broken cross-account inherit.
+
+  **Symptom of the cross-account bug**: `gh secret list` shows `SYNC_PAT` set and recent, but the workflow's `Check SYNC_PAT visibility` step logs `SYNC_PAT_PRESENT:` (empty). Fix: `aidevops sync-workflows --apply`.
+
 - **Referencing `@main` is a trust boundary.** You're trusting whoever controls aidevops's `main` branch. For higher-trust deployments, pin to a version tag (`@v3.9.0`) and update explicitly.
 - **`pull_request_target` vs `pull_request`.** The reusable workflow's job guards accept either event type. The caller picks based on its security model:
   - **Private repo with trusted contributors**: `pull_request` is simpler and has fewer footguns.
@@ -117,6 +140,20 @@ See also [`auto-dispatch.md`](auto-dispatch.md) for the `SYNC_PAT` requirement (
 - **`default_workflow_permissions: read` repos (GH#20967).** GitHub's recommended security default is `default_workflow_permissions: read`. Reusable workflow job-level `permissions:` declarations cannot exceed the CALLER's ceiling — they are capped at whatever the caller workflow grants. A caller with no `permissions:` block inherits the repo's restrictive default, so GitHub refuses to create any jobs (`conclusion: startup_failure`, zero jobs). The canonical caller templates include a top-level `permissions:` block that is the union of all job-level permissions used by the reusable workflow. If you add a new permission to a reusable workflow job, also update the matching caller template.
 
   Verification: `gh api repos/OWNER/REPO/actions/permissions/workflow --jq .default_workflow_permissions` returns `"read"` or `"write"`. A `"read"` repo will fail without the caller's `permissions:` block.
+
+## Framework self-test assumption guard
+
+Both GH#20967 (missing `permissions:` block) and GH#20976 (`secrets: inherit` cross-account failure) share the same root cause class: the canonical caller templates were authored against the framework's own self-test scenario (`marcusquinn/aidevops` calling itself) which has same-account and same-repo semantics that don't hold for downstream consumers.
+
+When authoring or reviewing a canonical caller template, apply this checklist to catch the pattern before it ships:
+
+| Check | Self-test passes? | Downstream breaks? | Gate |
+|---|---|---|---|
+| `secrets: inherit` for user-defined secrets | Yes (same-account) | Yes (cross-account empty) | Always use explicit `secrets: MySecret: ${{ secrets.MySecret }}` for user-defined secrets |
+| Missing `permissions:` block | Yes (`default_workflow_permissions: write` on framework repo) | Yes (`startup_failure` on read-default repos) | Always include a top-level `permissions:` block with the union of all job-level permissions |
+| Hardcoded `marcusquinn` org references | Yes (same repo) | Possible (wrong org) | Use `github.repository_owner` or `inputs:` for org-specific values |
+
+This checklist lives here so the next "framework self-test passes, downstream broken" instance surfaces during template authoring, not after a downstream user files a bug report.
 
 ## Migration: from copied workflow to caller
 
@@ -162,6 +199,8 @@ To make a new aidevops workflow reusable by downstream repos:
 - Issue [#20648](https://github.com/marcusquinn/aidevops/issues/20648) — Phase 1 drift detector
 - Issue [#20649](https://github.com/marcusquinn/aidevops/issues/20649) — Phase 2 opt-in resync
 - Issue [#20727](https://github.com/marcusquinn/aidevops/issues/20727) — review-bot-gate migration (SHA-pin stale drift)
+- Issue [#20967](https://github.com/marcusquinn/aidevops/issues/20967) — missing `permissions:` block in caller template (sister bug, same root cause class)
+- Issue [#20976](https://github.com/marcusquinn/aidevops/issues/20976) — `secrets: inherit` fails cross-account; canonical template switched to explicit pass-through
 - Issue [#21154](https://github.com/marcusquinn/aidevops/issues/21154) — maintainer-gate migration (layer-1 defense-in-depth propagation)
 - Reference [incident-gh17671-supply-chain.md](incident-gh17671-supply-chain.md) — postmortem that motivated maintainer-gate propagation
 - GitHub docs: [Reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows)

--- a/.agents/templates/workflows/issue-sync-caller.yml
+++ b/.agents/templates/workflows/issue-sync-caller.yml
@@ -55,6 +55,12 @@ on:
 jobs:
   sync:
     uses: marcusquinn/aidevops/.github/workflows/issue-sync-reusable.yml@main
-    secrets: inherit
+    # GH#20976: `secrets: inherit` only works within the same GitHub account/org.
+    # Cross-account consumers (every downstream user) receive empty values for any
+    # caller-repo secret when using `secrets: inherit` against a cross-account
+    # reusable workflow. Explicit pass-through is required for secrets to resolve.
+    # If a new secret is added to issue-sync-reusable.yml, add it here too.
+    secrets:
+      SYNC_PAT: ${{ secrets.SYNC_PAT }}
     with:
       command: ${{ inputs.command || '' }}

--- a/.agents/templates/workflows/loc-badge-caller.yml
+++ b/.agents/templates/workflows/loc-badge-caller.yml
@@ -34,7 +34,11 @@ on:
 jobs:
   badges:
     uses: marcusquinn/aidevops/.github/workflows/loc-badge-reusable.yml@main
-    secrets: inherit
+    # GH#20976: `secrets: inherit` only works within the same GitHub account/org.
+    # Cross-account consumers (every downstream user) must pass secrets explicitly.
+    # If a new secret is added to loc-badge-reusable.yml, add it here too.
+    secrets:
+      SYNC_PAT: ${{ secrets.SYNC_PAT }}
     # Optional input overrides — uncomment and edit as needed:
     # with:
     #   top_n: 8

--- a/.github/workflows/issue-sync-reusable.yml
+++ b/.github/workflows/issue-sync-reusable.yml
@@ -65,6 +65,8 @@ jobs:
       # t2166: emit a visible warning when SYNC_PAT is unset so operators
       # see on every run — not just after GH006 — that TODO.md push will
       # fall back to GITHUB_TOKEN and be rejected by branch protection.
+      # GH#20976: SYNC_PAT can also appear empty if the caller uses
+      # `secrets: inherit` against a cross-account reusable workflow.
       - name: Check SYNC_PAT visibility (t2166)
         if: steps.check-author.outputs.skip != 'true'
         env:
@@ -72,9 +74,9 @@ jobs:
           REPO: ${{ github.repository }}
         run: |
           if [[ -z "${SYNC_PAT_PRESENT:-}" ]]; then
-            echo "::warning::SYNC_PAT secret is not set — TODO.md auto-sync push will be rejected by branch protection (GH006)."
-            echo "::warning::Fix: gh secret set SYNC_PAT --repo ${REPO} --body \"<FINE_GRAINED_PAT>\" (Contents: Read and write)"
-            echo "::warning::See todo/tasks/t2166-brief.md for the SYNC_PAT setup walkthrough."
+            echo "::warning::SYNC_PAT not present in this run — TODO.md push will be rejected by branch protection (GH006)."
+            echo "::warning::Cause A (secret not set): gh secret set SYNC_PAT --repo ${REPO} --body '<FINE_GRAINED_PAT>' (Contents: Read and write)"
+            echo "::warning::Cause B (cross-account caller): your caller uses 'secrets: inherit' which fails cross-account — run: aidevops sync-workflows --apply (see GH#20976)"
           else
             echo "::notice::SYNC_PAT present — TODO.md push will use PAT (bypasses branch protection via enforce_admins:false)"
           fi
@@ -211,6 +213,8 @@ jobs:
       # t2166: SYNC_PAT visibility — this job also pushes TODO.md updates
       # on every issue-opened/closed event, so it hits the same GH006 when
       # SYNC_PAT is unset. Warn explicitly before doing any work.
+      # GH#20976: SYNC_PAT can also appear empty if the caller uses
+      # `secrets: inherit` against a cross-account reusable workflow.
       - name: Check SYNC_PAT visibility (t2166)
         if: steps.check-issue.outputs.sync == 'true'
         env:
@@ -218,8 +222,9 @@ jobs:
           REPO: ${{ github.repository }}
         run: |
           if [[ -z "${SYNC_PAT_PRESENT:-}" ]]; then
-            echo "::warning::SYNC_PAT secret is not set — TODO.md auto-sync push will be rejected by branch protection (GH006)."
-            echo "::warning::Fix: gh secret set SYNC_PAT --repo ${REPO} --body \"<FINE_GRAINED_PAT>\" (Contents: Read and write)"
+            echo "::warning::SYNC_PAT not present in this run — TODO.md push will be rejected by branch protection (GH006)."
+            echo "::warning::Cause A (secret not set): gh secret set SYNC_PAT --repo ${REPO} --body '<FINE_GRAINED_PAT>' (Contents: Read and write)"
+            echo "::warning::Cause B (cross-account caller): your caller uses 'secrets: inherit' which fails cross-account — run: aidevops sync-workflows --apply (see GH#20976)"
           else
             echo "::notice::SYNC_PAT present — TODO.md push will use PAT"
           fi
@@ -295,14 +300,17 @@ jobs:
 
     steps:
       # t2166: SYNC_PAT visibility for manual dispatch runs too.
+      # GH#20976: SYNC_PAT can also appear empty if the caller uses
+      # `secrets: inherit` against a cross-account reusable workflow.
       - name: Check SYNC_PAT visibility (t2166)
         env:
           SYNC_PAT_PRESENT: ${{ secrets.SYNC_PAT != '' && 'true' || '' }}
           REPO: ${{ github.repository }}
         run: |
           if [[ -z "${SYNC_PAT_PRESENT:-}" ]]; then
-            echo "::warning::SYNC_PAT secret is not set — any TODO.md push from this manual run will be rejected by branch protection (GH006)."
-            echo "::warning::Fix: gh secret set SYNC_PAT --repo ${REPO} --body \"<FINE_GRAINED_PAT>\" (Contents: Read and write)"
+            echo "::warning::SYNC_PAT not present in this run — any TODO.md push from this manual run will be rejected by branch protection (GH006)."
+            echo "::warning::Cause A (secret not set): gh secret set SYNC_PAT --repo ${REPO} --body '<FINE_GRAINED_PAT>' (Contents: Read and write)"
+            echo "::warning::Cause B (cross-account caller): your caller uses 'secrets: inherit' which fails cross-account — run: aidevops sync-workflows --apply (see GH#20976)"
           else
             echo "::notice::SYNC_PAT present — TODO.md push will use PAT"
           fi
@@ -391,14 +399,17 @@ jobs:
       # Ensures the warning/notice emits unconditionally on every merged PR,
       # not just when the downstream TODO.md proof-log path actually runs
       # (which is skipped by range-syntax titles and planning-only PR bodies).
+      # GH#20976: SYNC_PAT can also appear empty if the caller uses
+      # `secrets: inherit` against a cross-account reusable workflow.
       - name: Check SYNC_PAT visibility (t2166)
         env:
           SYNC_PAT_PRESENT: ${{ secrets.SYNC_PAT != '' && 'true' || '' }}
           REPO: ${{ github.repository }}
         run: |
           if [[ -z "${SYNC_PAT_PRESENT:-}" ]]; then
-            echo "::warning::SYNC_PAT secret is not set — TODO.md auto-sync push will be rejected by branch protection (GH006)."
-            echo "::warning::Fix: gh secret set SYNC_PAT --repo ${REPO} --body \"<FINE_GRAINED_PAT>\" (Contents: Read and write)"
+            echo "::warning::SYNC_PAT not present in this run — TODO.md auto-sync push will be rejected by branch protection (GH006)."
+            echo "::warning::Cause A (secret not set): gh secret set SYNC_PAT --repo ${REPO} --body '<FINE_GRAINED_PAT>' (Contents: Read and write)"
+            echo "::warning::Cause B (cross-account caller): your caller uses 'secrets: inherit' which fails cross-account — run: aidevops sync-workflows --apply (see GH#20976)"
           else
             echo "::notice::SYNC_PAT present — TODO.md push will use PAT"
           fi

--- a/.github/workflows/loc-badge-reusable.yml
+++ b/.github/workflows/loc-badge-reusable.yml
@@ -71,6 +71,8 @@ jobs:
             echo "skip=false" >> "$GITHUB_OUTPUT"
           fi
 
+      # GH#20976: SYNC_PAT can appear empty even when set on the caller if the
+      # caller uses `secrets: inherit` against a cross-account reusable workflow.
       - name: Check SYNC_PAT visibility
         if: steps.check-author.outputs.skip != 'true'
         env:
@@ -78,8 +80,9 @@ jobs:
           REPO: ${{ github.repository }}
         run: |
           if [[ -z "${SYNC_PAT_PRESENT:-}" ]]; then
-            echo "::warning::SYNC_PAT is not set — badge commit may be rejected by branch protection (GH006)."
-            echo "::warning::Fix: gh secret set SYNC_PAT --repo ${REPO} (Contents: Read and write)"
+            echo "::warning::SYNC_PAT not present in this run — badge commit may be rejected by branch protection (GH006)."
+            echo "::warning::Cause A (secret not set): gh secret set SYNC_PAT --repo ${REPO} (Contents: Read and write)"
+            echo "::warning::Cause B (cross-account caller): your caller uses 'secrets: inherit' which fails cross-account — run: aidevops sync-workflows --apply (see GH#20976)"
           else
             echo "::notice::SYNC_PAT present — badge commit will use PAT"
           fi

